src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -110,40 +123,69 @@ public sealed class Options
             public string LabelColumnName;
 
             /// <summary>
-            /// Loss function minimized for finding factor matrices.  Two values are allowed, 0 or 12. The values 0 means traditional collaborative filtering
-            /// problem with squared loss. The value 12 triggers one-class matrix factorization for implicit-feedback recommendation problem.
+            /// Loss function minimized for finding factor matrices.
             /// </summary>
+            /// <remarks>
+            /// Two values are allowed, <see cref="LossFunctionType.SquareLossRegression"/> or <see cref="LossFunctionType.SquareLossOneClass"/>.
+            /// The <see cref="LossFunctionType.SquareLossRegression"/> means traditional collaborative filtering problem with squared loss.
+            /// The <see cref="LossFunctionType.SquareLossOneClass"/> triggers one-class matrix factorization for implicit-feedback recommendation problem.
```

Comment:
Please add `For the first case, please see Equation (1) in [https://www.csie.ntu.edu.tw/~cjlin/papers/libmf/mf_adaptive_pakdd.pdf](https://www.csie.ntu.edu.tw/~cjlin/papers/libmf/mf_adaptive_pakdd.pdf) for the formulation considered. For the second case, please see Equation (3) in [http://yifanhu.net/PUB/cf.pdf`](http://yifanhu.net/PUB/cf.pdf`).

Reply:
I'll add that to loss functions

---
In reply to: [254911243](https://github.com/dotnet/machinelearning/pull/2451#discussion_r254911243) [](ancestors = 254911243)

Reply:
Can you check them as remarks on loss functions enum?

---
In reply to: [254911715](https://github.com/dotnet/machinelearning/pull/2451#discussion_r254911715) [](ancestors = 254911715,254911243)